dashboard/app/models/script.rb

```diff
@@ -596,13 +596,25 @@ def self.get_unit_family_redirect_for_user(family_name, user: nil, locale: 'en-U
 
     family_units = Script.get_family_from_cache(family_name).sort_by(&:version_year).reverse
 
-    # Only students should be redirected based on unit progress and/or section assignments.
-    if user&.student?
+    return nil unless family_units.last.can_be_instructor?(user) || family_units.last.can_be_participant?(user)
+
+    # Only signed in participants should be redirected based on unit progress and/or section assignments.
+    if user && family_units.last.can_be_participant?(user)
       assigned_unit_ids = user.section_scripts.pluck(:id)
       progress_unit_ids = user.user_levels.map(&:script_id)
       unit_ids = assigned_unit_ids.concat(progress_unit_ids).compact.uniq
       unit_name = family_units.select {|s| unit_ids.include?(s.id)}&.first&.name
-      return Script.new(redirect_to: unit_name, published_state: SharedCourseConstants::PUBLISHED_STATE.beta) if unit_name
+      if unit_name
+        # This creates a temporary script which is used to redirect the user. The audiences are set
+        # to instructor as teacher and student as participant because that will allow the redirect to
+        # happen for any course
```

Comment:
> The audiences are set to instructor as teacher and student as participant"

I'm having trouble understanding this part of the comment. I think the code says clearly enough what it is doing so I would leave this part out and just say why it is doing it, e.g. "The audiences are set to allow..."

I think I am missing something though. Why does this strategy allow redirects for any course? would it make more sense to say that this allows users of any type to be redirected?

Reply:
I have updated the comment based on your feedback. Thanks!